deploy/helm/kuberhealthy/templates/khcheck-daemonset.yaml

```diff
@@ -8,6 +8,14 @@ metadata:
 spec:
   runInterval:  {{ .Values.check.daemonset.runInterval }}
   timeout: {{ .Values.check.daemonset.timeout }}
+  {{- if .Values.check.daemonset.extraAnnotations }}
+  extraAnnotations:
```

Comment:
Could we have instead a helper function instead of direct use of the value?

The main benefit, is we could override behaviour for extra annotations and extra labels

https://github.com/kuberhealthy/kuberhealthy/blob/master/deploy/helm/kuberhealthy/templates/_helpers.tpl

{{- define "kuberhealthy.extra-labels" -}}
{{- define "kuberhealthy.extra-annotations" -}}

Reply:
hm, I don't think it would work, since these values are set per predefined checks (daemonset, deployment, dns-external, etc) and I would prefer to be able to set these per check and not per chart (unless I'm missing something obvious here)

I could implement a `globalExtraAnnotations` and `globalExtraLabels` but that was not my goal with this PR.

Reply:
@ivankatliarchuk do you agree with my comment? how do you see this being implemented?

Reply:
And another point. 
Example

```tpl
 {{- if .Values.check.daemonset.extraAnnotations }}
  extraAnnotations:
    {{- toYaml .Values.check.daemonset.extraAnnotations | nindent 4 }}
  {{- end }}
```

This means annotations need to be set per check, when is nice to have some sort of global annotations

```
  extraAnnotations:
    {{- toYaml .Values.check.annotations | nindent 4 }}
    {{- toYaml .Values.check.daemonset.extraAnnotations | nindent 4 }}   
```

Not sure if this supported

Reply:
@ivankatliarchuk okay, templating was implemented, but not the global annotations. I bumped into a problem while trying to implement the global annotation which would make the solution not clean.

```tpl
{{- define "kuberhealthy.extraAnnotations" }}
  {{- if . -}}
  extraAnnotations:
    {{- toYaml .Values.check.extraAnnotations | nindent 2 }}
    {{- toYaml . | nindent 2 }}
  {{- end -}}
{{ end -}}
```

^ this wouldn't work because the template doesn't have the full context (`.Values`) loaded on it and passing it as an extra arg to all templating calls would create the same problem that we were trying to avoid with the other method. I could add if you see the value on it 😄

Reply:
@ivankatliarchuk I would love to have your feedback here. My kuberhealthy checks keep failing with false-positives because of karpenter killing the nodes before the kuberhealthy checks are done and adding the annotation would fix it

Reply:
@integrii @jandersen-plaid @nak0f sorry for tagging you, but it has been hard to communicate with @ivankatliarchuk about this issue

Reply:
@ivankatliarchuk Have you checked the last commit? I've implemented the templating; it's just that I couldn't achieve what you suggested.